src/main/java/com/o19s/es/ltr/feature/store/index/IndexFeatureStore.java

```diff
@@ -94,12 +94,12 @@
     }
 
     private final String index;
-    private final Client client;
+    private final QueryShardContext context;
```

Comment:
we could type this as `Supplier<Client> clientSupplier`, depending on QueryShardContext is not really necessary

Reply:
Great! Done!